internal/agent/install.go

```diff
@@ -213,9 +213,10 @@ func RunInstall(c *config.Config) error {
 	utils.SetEnv(c.Env)
 	utils.SetEnv(c.Install.Env)
 
-	if c.Install.Device == "" || c.Install.Device == "auto" {
-		c.Install.Device = detectDevice()
-	}
+	// if c.Install.Device == "" || c.Install.Device == "auto" {
+	// 	c.Install.Device = detectDevice()
+	// }
+	c.Install.Device = "/dev/vdb"
```

Comment:
we need to respect `NoFormat: true` and detect the target device using labels. `detectDevice()` will simply point to `/dev/vda` (in my example in the PR description) which is not correct.

We should find the device that has the needed labels, validate that all needed partitions are there and if true, make that the target device.

Reply:
Now there is `DetectPreConfiguredDevice` which looks for a disk with the appropriate labels.